From 5520861ac4bb11bf683309cc7d07c5722774187e Mon Sep 17 00:00:00 2001
From: PulkoMandy <pulkomandy@pulkomandy.tk>
Date: Sat, 22 Feb 2025 22:39:29 +0100
Subject: Adjust for Haiku

- Link against libbsd for cfsetspeed
- Remove FNM_EXTMATCH (unsupported in Haiku)
- Adjust name of m68k C compiler

diff --git a/Makefile b/Makefile
index 4111755..7daa2b5 100644
--- a/Makefile
+++ b/Makefile
@@ -4,11 +4,11 @@ export CXX = g++
 export AR = ar rc
 export RANLIB = ranlib
 export STRIP = strip
-export LIBS =
+export LIBS = -lbsd
 
-export CC68K = m68k-linux-gnu-gcc
-export LD68K = m68k-linux-gnu-ld
-export OBJCOPY68K = m68k-linux-gnu-objcopy
+export CC68K = m68k-elf-gcc
+export LD68K = m68k-elf-ld
+export OBJCOPY68K = m68k-elf-objcopy
 
 export C68KFLAGS = -g -Os -fomit-frame-pointer -march=68000
 export OBJDIR = .obj
diff --git a/src/mc68328.cc b/src/mc68328.cc
index 77f7ae8..e4cd165 100644
--- a/src/mc68328.cc
+++ b/src/mc68328.cc
@@ -173,7 +173,7 @@ bool foreach_register(const char* glob, std::function<bool(const std::string& na
 {
 	for (const auto& i : registers)
 	{
-		if (fnmatch(glob, i.first.c_str(), FNM_CASEFOLD|FNM_EXTMATCH) == 0)
+		if (fnmatch(glob, i.first.c_str(), FNM_CASEFOLD) == 0)
 		{
 			if (cb(i.first, i.second))
 				return true;
-- 
2.45.2

